build.gradle

```diff
@@ -273,7 +325,57 @@ class PreDist extends DefaultTask {
         project.fileTree(dir: 'build/generated-dist', include: '**/*${group}*/**').each { file ->
             file.renameTo(project.file(file.getAbsolutePath().replace('${group}', "${groupName}")))
         }
-
+        //docbook unzip and put only necessary files to generated-dist
+		project.copy {
```

Comment:
All extraction should be done in `extractDependencies` task. `preDist` is specifically for copying from the cache into the directory structure desired for distribution. In other words `extractDependencies` can only write to `build/dependency-cache` and `preDist` can only write to `build/generated-dist`. Motivation for the clear separation is both for developer reasoning and Gradle incremental building.

Reply:
Move the extraction from preDist to extractDependencies.